### PR TITLE
Make keyboard shortcuts for comments work when pages are rotated

### PR DESCRIPTION
**Pull request by mdbenjam**
Connects: #2522 

This is the final part of rotating pages. This allows users to control placing the comment with the keyboard even while a page is rotated.

**Test Plan**
- [ ] Rotate a document by 90, 180, and 270 degrees, then add a comment using the keyboard (alt + c, then use arrow keys), make sure the arrow keys move the comment in the correct direction.

### CODE REVIEW

File: client/app/reader/PdfViewer.jsx [hunk 1]
```diff
@@ -14,15 +14,20 @@ import { getFilteredDocuments } from './selectors';
 import * as Constants from '../reader/constants';
 import { CATEGORIES, ACTION_NAMES, INTERACTION_TYPES } from '../reader/analytics';
 
-export const getNextAnnotationIconPageCoords = (direction, placingAnnotationIconPageCoords, pages, file) => {
+const NUMBER_OF_DIRECTIONS = 4;
+
+export const getNextAnnotationIconPageCoords = (direction, placingAnnotationIconPageCoords, pages, file, rotation) => {
```
Comment (amprokop) [inline client/app/reader/PdfViewer.jsx]: Do you think it'd be a good idea to add a little synopsis of how this is used? Especially in the really complicated PDF viewing part, I'd be in favor of having more comments to read before diving into a function.

Comment (amprokop) [reply client/app/reader/PdfViewer.jsx]: I think this gets the new coordinates after the user moves the mouse or keyboard for comment icons, but as someone who is not familiar with the codebase it'd be nice to have 100% certainty before diving in.

Comment (amprokop) [reply client/app/reader/PdfViewer.jsx]: Just a suggestion though — the naming all over these files is generally excellent IMO.

Comment (mdbenjam) [reply client/app/reader/PdfViewer.jsx]: That's a super valid concern. I'll add some comments now!

File: client/app/reader/PdfViewer.jsx [hunk 2]
```diff
@@ -14,15 +14,25 @@ import { getFilteredDocuments } from './selectors';
...
+
+// Given a direction, the current coordinates, an array of the div elements for each page,
+// the file, and rotation of the document, this function calculates the next location of the comment.
+export const getNextAnnotationIconPageCoords = (direction, placingAnnotationIconPageCoords, pages, file, rotation) => {
+  // There are four valid rotations: 0, 90, 180, 270. We transform those values to 0, -1, -2, -3.
+  // We then use that value to rotate the direction. I.E. Hitting up (value 0) on the
+  // keyboard when rotated 90 degrees corresponds to moving left (value 3) on the document.
+  const rotationIncrements = -(rotation / Constants.ROTATION_INCREMENTS) % NUMBER_OF_DIRECTIONS;
+  const transformedDirection = Constants.MOVE_ANNOTATION_ICON_DIRECTION_ARRAY[
+    (direction + rotationIncrements + NUMBER_OF_DIRECTIONS) % NUMBER_OF_DIRECTIONS];
```
Comment (NickHeiner) [inline client/app/reader/PdfViewer.jsx]: This array index is sufficiently complicated that I think it deserves to be extracted into its own var.

```js
(direction + rotationIncrements + NUMBER_OF_DIRECTIONS) % NUMBER_OF_DIRECTIONS
```

File: client/app/reader/PdfViewer.jsx [hunk 3]
```diff
@@ -14,15 +14,25 @@ import { getFilteredDocuments } from './selectors';
 import * as Constants from '../reader/constants';
 import { CATEGORIES, ACTION_NAMES, INTERACTION_TYPES } from '../reader/analytics';
 
-export const getNextAnnotationIconPageCoords = (direction, placingAnnotationIconPageCoords, pages, file) => {
+const NUMBER_OF_DIRECTIONS = 4;
+
+// Given a direction, the current coordinates, an array of the div elements for each page,
+// the file, and rotation of the document, this function calculates the next location of the comment.
+export const getNextAnnotationIconPageCoords = (direction, placingAnnotationIconPageCoords, pages, file, rotation) => {
+  // There are four valid rotations: 0, 90, 180, 270. We transform those values to 0, -1, -2, -3.
```
Comment (NickHeiner) [inline client/app/reader/PdfViewer.jsx]: Is it odd that the values are negative here, but positive in `Constants` below?

**Comment by NickHeiner** [comment]
I didn't do a full investigation but generally LGTM.

File: client/app/reader/PdfViewer.jsx (hunk 3 above)
Comment (mdbenjam) [reply client/app/reader/PdfViewer.jsx]: We have to rotate the opposite direction. Counter-clockwise to make up for the clockwise rotation. I think it makes sense to leave the constants making sense in a clockwise rotation.

**Pull request merged**